freetextresponse/private/view.js

```diff
@@ -4,46 +4,109 @@ function FreeTextResponseView(runtime, element) {
                 usedAttemptsFeedback.text(response.used_attempts_feedback);
-                submitParent.removeClass();
-                submitParent.addClass(response.submit_class);
+                buttonSubmit.addClass(response.nodisplay_class);
                 problemProgress.text(response.problem_progress);
-                wordCountError.text(response.word_count_message);
                 submissionReceivedMessage.text(response.submitted_message);
                 buttonSubmit.text('Submit');
-
+                userAlertMessage.text(response.user_alert);
+                buttonSave.addClass(response.nodisplay_class);
+                // Update Indicators
+                if ( '' === response.visibility_class ) {
+                    textareaParent.removeClass('correct');
+                    textareaParent.removeClass('incorrect');
+                    textareaParent.removeClass('unanswered');
+                    textareaParent.addClass(response.indicator_class); 
+                }
+                
                 runtime.notify('submit', {
                     state: 'end'
                 });
             },
-            error: function buttonSubmitOnError() {
+            error: function buttonSubmitOnError(err) {
                 runtime.notify('error', {});
             }
         });
         return false;
     });
+
+    buttonSave.on('click', function () {
+        buttonSave.text('Checking...');
+        runtime.notify('save', {
+            message: 'Saving...',
+            state: 'start'
+        });
+        $.ajax(urlSave, {
+            type: 'POST',
+            data: JSON.stringify({
+                'student_answer': $element.find('.student_answer').val()
+            }),
+            success: function buttonSaveOnSuccess(response) {
+                buttonSubmit.addClass(response.nodisplay_class);
+                buttonSave.addClass(response.nodisplay_class);
+                usedAttemptsFeedback.text(response.used_attempts_feedback);
+                problemProgress.text(response.problem_progress);
+                submissionReceivedMessage.text(response.submitted_message);
+                buttonSave.text('Save');
+                userAlertMessage.text(response.user_alert);
+                // Update Indicators
+                if ( '' === response.visibility_class ) {
+                    textareaParent.removeClass('correct');
+                    textareaParent.removeClass('incorrect');
+                    textareaParent.removeClass('unanswered');
+                    textareaParent.addClass(response.indicator_class); 
+                }
+                
+                runtime.notify('save', {
+                    state: 'end'
+                });
+            },
+            error: function buttonSaveOnError() {
+                runtime.notify('error', {});
+            }
+        });
+        return false;
+    });
+
+    textareaStudentAnswer.on('keydown', function() {
```

Comment:
Oh, Your talking about a space between the function and the parens.   It would have been more clear to explicitly write that out like, "you're missing a space between the function keyword and it's parentheses."  Fixed